### PR TITLE
[Azure][Container_registry] Add dimension and metric_type metadata to the container_registry datastream

### PR DESCRIPTION
**Pull request by tetianakravchenko**
## What does this PR do?

- remove duplicated fields
- instead of using wildcard in the field name - use predefined field names
    - for azure.dimensions - there are no dimensions
    > Note: only 1 metric - `StorageUsed` has a dimension, but we do not use it https://learn.microsoft.com/en-us/azure/azure-monitor/reference/supported-metrics/microsoft-containerregistry-registries-metrics
    - list of metrics - https://learn.microsoft.com/en-us/azure/azure-monitor/reference/supported-metrics/microsoft-containerregistry-registries-metrics

- used fields for dimensions:
   - `cloud.region`
   - `azure.resource.id` - this field includes azure.subscription_id, azure.resource.group, azure.resource.name, format:
       ```
        /subscriptions/{azure.subscription_id}/resourceGroups/{azure.resource.group}providers/Microsoft.ContainerRegistry/registries/{azure.resource.name}
      ```

   - `azure.timegrain`
       > NOTE: there is an issue with `timegrain` as it does not work as expected, I will create an issue and link it here.

- metrics - use predefined metrics - since in the configuration is defined list of metrics

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Comment by zmoog** [comment]
@tetianakravchenko I see https://github.com/elastic/integrations/issues/7162 it is the tracking issue for the `timegrain` problem.

Do you see this problem as a blocker for this PR?

**Review by zmoog — APPROVED** [review]

**Comment by agithomas** [comment]
Which field will hold the names of the images, tags, other meta data information ? Shouldn't they be dimensions?

**Comment by tetianakravchenko** [comment]
@agithomas 
> Which field will hold the names of the images, tags, other meta data information ? Shouldn't they be dimensions?

There is no such information available in this datastream, here is the list of metrics - https://learn.microsoft.com/en-us/azure/azure-monitor/reference/supported-metrics/microsoft-containerregistry-registries-metrics

as you see those metrics are on the registry level, not for each specific image

all available fields:
Screenshot 2023-08-02 at 12 41 53

**Review by agithomas — APPROVED** [review]
LGTM!

File: packages/azure_metrics/data_stream/container_registry/fields/fields.yml [hunk 1]
```diff
@@ -1,7 +1,35 @@
...
+      description: Number of image pushes in total
+    - name: run_duration.total
+      type: float
+      metric_type: gauge
+      unit: ms
+      description: ACR tasks run duration in milliseconds
+    - name: agent_pool_cpu_time.total
+      type: float
+      metric_type: gauge
+      unis: s
```
Comment (ritalwar) [inline packages/azure_metrics/data_stream/container_registry/fields/fields.yml:28]: > Typo: "unis" should be "unit".

**Pull request merged**